src/java.base/share/classes/sun/security/ssl/JsseJce.java

```diff
@@ -165,7 +207,24 @@ private static class EcAvailability {
         static {
             boolean mediator = true;
             try {
-                Signature.getInstance(SIGNATURE_ECDSA);
+                try {
+                    Signature.getInstance(SIGNATURE_ECDSA);
+                } catch (NoSuchAlgorithmException e) {
+                    // When running in FIPS mode, the signature SHA1withECDSA is not
+                    // available by default. In this scenario we should still set EC
+                    // availability to true since other algorithms in the ECDSA signature
+                    // family are available for use in various ECDSA TLS ciphers. All
+                    // FIPS solutions are expected to have the algorithm "SHA512withECDSA"
+                    // so we will check for this instead when running in FIPS mode.
+                    if (!(isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_224) ||
+                          isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_256) ||
+                          isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_384) ||
+                          isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_512)
+                         )) {
+                            throw NoSuchAlgorithmException("No expected ECDSA algorithms found.");
+                    }
+                }
+
```

Comment:
There's no need to create a new exception to throw, nor even to throw at all, and the try block is not needed with the new helper method; suggest:
```java
                mediator = isSignatureAlgorithmAvailable(SIGNATURE_ECDSA)
                        || isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_224)
                        || isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_256)
                        || isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_384)
                        || isSignatureAlgorithmAvailable(SIGNATURE_ECDSA_512);
```
Please update the comment to reflect that we're not only relying on `SHA512withECDSA` for backup.

Reply:
Yes this reads better. I updated the code and comment accordingly.